packages/es-ingest/src/main/java/software/uncharted/terarium/esingest/ingests/IElasticIngest.java

```diff
@@ -0,0 +1,18 @@
+package software.uncharted.terarium.esingest.ingests;
+
+import software.uncharted.terarium.esingest.models.input.IInputDocument;
+import software.uncharted.terarium.esingest.models.input.IInputEmbeddingChunk;
+import software.uncharted.terarium.esingest.models.output.IOutputDocument;
+import software.uncharted.terarium.esingest.models.output.IOutputEmbeddingChunk;
+
+public interface IElasticIngest<DocInputType extends IInputDocument, DocOutputType extends IOutputDocument, EmbeddingInputChunkType extends IInputEmbeddingChunk, EmbeddingOutputChunkType extends IOutputEmbeddingChunk> {
+
+	public abstract DocOutputType processDocument(DocInputType input);
```

Comment:
Strictly speaking I don't think `abstract` is needed here because this is an `interface` not a `class`